packages/components/src/hooks/use-hover.ts

```diff
@@ -13,26 +14,64 @@ export const useHover = <T extends HTMLElement & SVGSVGElement>(
 
     React.useEffect(() => {
         const node = ref.current;
-        if (node) {
-            if (should_prevent_bubbling) {
+
+        if (!node) return;
+
+        const addMouseEnterEvent = () => {
+            if (isMobile) {
+                node.addEventListener('touchstart', handleHoverBegin);
+                node.addEventListener('touchend', handleHoverFinish);
+            } else {
```

Comment:
My man, so many tings are repeated. I think we can compress
```suggestion
React.useEffect(() => {
    const node = ref.current;

    if (!node) return;

    const handleEventBegin = () => {
        if (isMobile) {
            node.addEventListener('touchstart', handleHoverBegin);
            node.addEventListener('touchend', handleHoverFinish);
        } else {
            node.addEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
            node.addEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
        }
    };

    const handleEventFinish = () => {
        if (isMobile) {
            node.removeEventListener('touchstart', handleHoverBegin);
            node.removeEventListener('touchend', handleHoverFinish);
        } else {
            node.removeEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
            node.removeEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
        }
    };

    handleEventBegin();

    return handleEventFinish;
}, [ref, should_prevent_bubbling]);
```

Reply:
This may work as well 🤔 
```suggestion
React.useEffect(() => {
    const node = ref.current;

    if (!node) return;

    const eventType = isMobile ? 'touch' : should_prevent_bubbling ? 'mouseenter' : 'mouseover';
    const eventFinishType = isMobile ? 'touchend' : should_prevent_bubbling ? 'mouseleave' : 'mouseout';

    const addEvent = (type, handler) => node.addEventListener(type, handler);
    const removeEvent = (type, handler) => node.removeEventListener(type, handler);

    const handleEventBegin = () => {
        addEvent(`${eventType}start`, handleHoverBegin);
        addEvent(eventFinishType, handleHoverFinish);
    };

    handleEventBegin();

    return () => {
        removeEvent(`${eventType}start`, handleHoverBegin);
        removeEvent(eventFinishType, handleHoverFinish);
    };
}, [ref, should_prevent_bubbling]);
```